src/v/kafka/server/replicated_partition.cc

```diff
@@ -485,9 +485,11 @@ replicated_partition::get_leader_epoch_last_offset_unbounded(
 ss::future<error_code> replicated_partition::prefix_truncate(
   model::offset kafka_truncation_offset,
   ss::lowres_clock::time_point deadline) {
-    if (
-      kafka_truncation_offset <= start_offset()
-      || kafka_truncation_offset > high_watermark()) {
+    if (kafka_truncation_offset <= start_offset()) {
```

Comment:
nit: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/cluster/Partition.scala#L1688

suggests that if truncation_offset < 0 it should throw a OffsetOutOfRangeException(), its an edge case so probably not a big deal.

Reply:
Ah, there is even more nuance here than that: `-1` indicates a complete wipe up to `high_watermark`, while other values `< 0` are an `offset_out_of_range` error. Will have this covered in a follow up :+1: